src/front/glsl/parser/types.rs

```diff
@@ -219,11 +219,11 @@ impl<'source> ParsingContext<'source> {
 
                     qualifiers.precision = Some((p, token.meta));
                 }
-                TokenValue::StorageAccess(access) => {
+                TokenValue::MemoryQualifier(access) => {
                     let storage_access = qualifiers
                         .storage_acess
-                        .get_or_insert((crate::StorageAccess::empty(), Span::default()));
-                    if storage_access.0.contains(access) {
+                        .get_or_insert((crate::StorageAccess::all(), Span::default()));
+                    if !storage_access.0.contains(!access) {
```

Comment:
wow, that's a bit strange of a check we have here.
How about `if !access.contains(storage_access.0)` instead?

Reply:
Won't work because if you have for example a single `writeonly` qualifier, it will try to check if `STORE.contains(LOAD | STORE)` which will be false and the branch will be taken producing an error.

Reply:
The check should be equivalent to `storage_access.0 | access != StorageAccess::all()`, though I'll admit I don't fully understand how that condition is connected with a memory qualifier being repeated

Reply:
This check is done to see if a memory qualifier was already used, as there are only two configurations `LOAD` or `STORE` and we `and` the storage access with the memory qualifier, the only way to check (without adding extra structures) is to check if the complement of the memory qualifier isn't present.

For example:
```
writeonly writeonly

Start:
- storage_access: all

First writeonly
- storage_access: all
- !STORE = LOAD
- storage_access.contains(LOAD) = true
- storage_access &= access <=> LOAD
 
Second writeonly
- storage_access: LOAD
- !STORE = LOAD
- storage_access.contains(LOAD) = false
- throw error
```